autowarefoundation/autoware.universe
# feat: add traffic signals merge node

## Description

Add converter node to change the traffic light interface between perception and planning.
This is stop 1 of https://github.com/autowarefoundation/autoware.universe/issues/2567#issuecomment-1461897576.

## Related links

https://github.com/autowarefoundation/autoware.universe/issues/2567


## Tests performed

1. Launch planning simulation and use `TrafficLightPublishPanel` to publish topic from TLR module.
2. Launch this package
`ros2 launch traffic_light_selector traffic_light_selector.launch.xml`
3. Check merged topic 
`ros2 topic echo /perception/traffic_light_selector/traffic_signals`

## Notes for reviewers



## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
